code_examples/sdk_examples/src/core_features/did/00_generate_did_keys.ts

```diff
@@ -0,0 +1,39 @@
+import { mnemonicGenerate, mnemonicToMiniSecret } from '@polkadot/util-crypto'
+
+import * as Kilt from '@kiltprotocol/sdk-js'
+
+export function generateKeypairs(mnemonic = mnemonicGenerate()): {
+  authentication: Kilt.KiltKeyringPair
+  attestation: Kilt.KiltKeyringPair
+  delegation: Kilt.KiltKeyringPair
+  encryption: Kilt.KiltEncryptionKeypair
+} {
+  // At first we generate a signer keypair and an encrytion keypair from the mnemonic
+  const baseSignerKey = Kilt.Utils.Crypto.makeKeypairFromSeed(
+    mnemonicToMiniSecret(mnemonic)
+  )
+  const encryptionKey = Kilt.Utils.Crypto.makeEncryptionKeypairFromSeed(
+    mnemonicToMiniSecret(mnemonic)
+  )
+
+  // Now we derive the authentication, attestation and delegation keypairs from the signer keypair
+  // by using different derivation paths.
+  // This allows us to restore all keys related to this DID from a single mnemonic seed phrase.
+  // The chosen derivation pathes can be anything, but here we use the purpuse of the keys to be generated.
+  const authenticationKey = baseSignerKey.derive(
+    '//authentication'
```

Comment:
Should we follow the way Sporran does it?

Reply:
Yes, I was about to update this. This would mean that our current accounts and DIDs used in the testing would break but hopefully fix itself.

Reply:
Though we already have this in the `utils` folder in the `core_features`. Therefore, I would take over this PR and sort it out.

Reply:
Maybe it is good to make it consistent with [the workshop](https://pr-223.d3oeht7cvddn19.amplifyapp.com/docs/develop/workshop/attester/did#create-a-did), that is already using the derivation paths from sporran.